emily/handler/tests/integration/deposit.rs

```diff
@@ -548,3 +548,81 @@ async fn update_deposits_updates_chainstate() {
         );
     }
 }
+
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn overwrite_deposit() {
```

Comment:
Once we add the reject also for the other states, can we add test cases for those as well? Other than the `Fullfilment`, everything else can be the same I guess